src/r/recordFactories/ProjectFactory.ts

```diff
@@ -227,97 +174,47 @@ export class ProjectFactory extends RecordFactory<RT.project> {
     }
   }
 
-  /**
-   * Ideally elements should get deleted via SceneFactory. But because we want deletion of media_upload element to delete the linked
-   * variable, we do this via ProjectFactory (as only ProjectFactory has access to variables).
-   */
-  deleteSceneSubRecord<N extends RT>(this: ProjectFactory, sceneId: number, type: N, id: number): RecordNode<N> | undefined {
+  deleteSceneDeepRecord<N extends RT>(this: ProjectFactory, sceneId: number, type: N, id: number): RecordNode<N> | undefined {
     const sceneJson = this.getRecord(RT.scene, sceneId);
-    if(sceneJson !== undefined) {
+    if (sceneJson !== undefined) {
       const record = (new SceneFactory(sceneJson)).deleteDeepRecord(type, id);
-      if(record?.type === RT.element) {
-        const currentRecord = (record as RecordNode<RT.element>);
-
-        switch(currentRecord.props.element_type) {
-          case ElementType.media_upload: {
-            const linkedVarId = currentRecord.props.media_upload_var_id as number;
-            this.deleteRecord(RT.variable, linkedVarId);
-            break;
-          }
-
-          case ElementType.embed_scorm: {
-            const linkedScoreVarId = currentRecord.props.embed_scorm_score_var_id as number;
-            this.deleteRecord(RT.variable, linkedScoreVarId);
-
-            const linkedSuspedDataVarId = currentRecord.props.embed_scorm_suspend_data_var_id as number;
-            this.deleteRecord(RT.variable, linkedSuspedDataVarId);
-
-            const linkedProgressVarId = currentRecord.props.embed_scorm_progress_var_id as number;
-            this.deleteRecord(RT.variable, linkedProgressVarId);
-
-            break;
-          }
-        }
+      if (record !== undefined) {
+        this.deleteLinkedVariables([ record as RecordNode<N> ]);
+        return record; 
       }
-      return record;
     }
+    return;
   }
 
-  /**
-   * Ideally elements should get duplicated via SceneFactory. But because we want duplication of media_upload element to duplicate the linked
-   * variable, we do this via ProjectFactory (as only ProjectFactory has access to variables).
-   * 
-   * NOTE: duplicateDeepRecord calls addDeepRecord internally.
-   * So why even override duplicate? (Add should take care of the additional conditions.)
-   * Because SceneFactory's duplicate calls SceneFactory's add,
-   * but our additional conditions are added to ProjectFactory's duplicateDeepRecord, which does NOT call addSceneSubRecord. 
-   */
-  duplicateSceneSubRecord<N extends RT>(this: ProjectFactory, sceneId: number, type: N, id: number): RecordNode<N> | undefined {
+  duplicateSceneDeepRecord<N extends RT>(this: ProjectFactory, sceneId: number, type: N, id: number): RecordNode<N> | undefined {
```

Comment:
Add documentation here on what's happening in this function and how it is different from before.